### PR TITLE
feat: move business logic out of SwirldsPlatform.java

### LINKED ISSUE #13580: Move business logic out of SwirldsPlatform
Move as much business logic as possible out of `SwirldsPlatform.java`. This class is too large and complex, and if there is business logic we need, that logic should be encapsulated in helper classes.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13580

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+     */
+    public ReconnectStateLoader(
+            @NonNull final Platform platform,
+            @NonNull final PlatformContext platformContext,
+            @NonNull final PlatformWiring platformWiring,
+            @NonNull final SwirldStateManager swirldStateManager,
+            @NonNull final SignedStateNexus latestImmutableStateNexus,
+            @NonNull final SavedStateController savedStateController,
+            @NonNull final AddressBook addressBook) {
+        this.platform = platform;
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java:81]: missing `Objects.requireNonNull` for all these args

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                                + reconnectHash + ", new hash is "
+                                + signedState.getState().getHash());
+            }
+
+            // Before attempting to load the state, verify that the platform AB matches the state AB.
+            AddressBookUtils.verifyReconnectAddressBooks(addressBook, signedState.getAddressBook());
+
+            swirldStateManager.loadFromSignedState(signedState);
+            // kick off transition to RECONNECT_COMPLETE before beginning to save the reconnect state to disk
+            // this guarantees that the platform statusp will be RECONNECT_COMPLETE before the state is saved
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java:124]: `statusp`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/StateInitializer.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    private static final Logger logger = LogManager.getLogger(StateInitializer.class);
+
+    private StateInitializer() {}
+
+    /**
+     * Initialize the state.
+     *
+     * @param signedState the state to initialize
+     */
+    public static void initializeState(
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/StateInitializer.java:55]: missing some param docs

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java:81]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ReconnectStateLoader.java:124]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/StateInitializer.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/StateInitializer.java:55]: added

**Review by alittley — APPROVED** [review]

**Pull request merged**